src/public/Invoke-Task.ps1

```diff
@@ -111,8 +111,35 @@ function Invoke-Task {
                                 & $task.PostAction
                             }
                         }
+                    } catch {
+                        # want to catch errors here _before_ we invoke TaskTearDown

```

Comment:
We're duplicating this error formatting logic which is already present [here](https://github.com/psake/psake/blob/master/src/public/Invoke-psake.ps1#L318). I think this is OK for now but we'll want to factor this out into a private function and DRY it up.

Reply:
I can take a crack at this today.

Reply:
- [x] refactored the common logic out of `Invoke-Psake` and `Invoke-Task` into a new `FormatErrorMessage` function;

I'm not 100% certain of the naming of this function (or the naming of the various error-related properties added to `Task`) as I'm not yet very familiar with the Powershell `ErrorRecord` and how to use it.

Regarding that new function, `FormatErrorMessage` seems to be a descriptive name given what it is doing, but since it calls into a version of Jeffery Snover's [Resolve-Error](https://devblogs.microsoft.com/powershell/resolve-error/) function I wonder if the relationship and distinction between `FormatErrorMessage` and `Resolve-Error` is clear enough 🤔